src/Collections/Sequences/Sorting.dfy

```diff
@@ -0,0 +1,181 @@
+// RUN: %dafny /compile:0 "%s"
+
+include "../../StandardLibrary.dfy"
+include "../../UInt.dfy"
+include "../Sequences/Seq.dfy"
+
+module sorting{
+  export
+    reveals Reflexive, AntiSymmetric, Connected, TotalOrdering
+    reveals LexicographicByteSeqBelow, LexicographicByteSeqBelowAux
+    provides AboutLexicographicByteSeqBelow
+    provides SortedBy, MergeSortBy, MergeSortedBy
+    provides StandardLibrary, UInt
+
+  import StandardLibrary
+  import opened UInt = StandardLibrary.UInt
+  import opened Seq
+
+predicate Reflexive<T(!new)>(R: (T, T) -> bool) {
```

Comment:
This predicate and a bunch of those after it are duplicated between here and the `BinarySearch` module. Maybe they should go in `StandardLibrary.dfy` until we reorganize it to split it up a little more?